lib/internal/source_map/source_map_cache.js

```diff
@@ -310,22 +314,25 @@ function findSourceMap(sourceURL) {
   if (!SourceMap) {
     SourceMap = require('internal/source_map/source_map').SourceMap;
   }
-  let sourceMap = esmSourceMapCache.get(sourceURL) ?? generatedSourceMapCache.get(sourceURL);
-  if (sourceMap === undefined) {
+  let entry = esmSourceMapCache.get(sourceURL) ?? generatedSourceMapCache.get(sourceURL);
+  if (entry === undefined) {
     for (const value of getCjsSourceMapCache()) {
       const filename = ObjectGetValueSafe(value, 'filename');
       const cachedSourceURL = ObjectGetValueSafe(value, 'sourceURL');
       if (sourceURL === filename || sourceURL === cachedSourceURL) {
-        sourceMap = {
-          data: ObjectGetValueSafe(value, 'data')
-        };
+        entry = value;
       }
     }
   }
-  if (sourceMap && sourceMap.data) {
-    return new SourceMap(sourceMap.data);
+  if (entry === undefined) {
+    return undefined;
+  }
+  let sourceMap = ObjectGetValueSafe(entry, 'sourceMap');
+  if (sourceMap === undefined) {
+    sourceMap = new SourceMap(ObjectGetValueSafe(entry, 'data'));
+    ObjectAssign(entry, { __proto__: null, sourceMap });
```

Comment:
What's the reasoning behind using `Object.assign` here?
```suggestion
    entry.sourceMap = sourceMap;
```

Reply:
Nit: From my personal experience, Object.assign is super slow.

Reply:
The key `sourceMap` is not present in the `entry` in the first place. As `ObjectAssign` is not a safe method, I've set the prototype of entries to `null` to avoid invoking unexpected setters.